From: Michal Hocko <mhocko@suse.cz> 
Subject: kabi: patches.fixes/mm-make-page-pfmemalloc-check-more-robust.patch
Patch-mainline: never
References: bnc#920016

page::pfmemalloc has been removed in patches.fixes/mm-make-page-pfmemalloc-check-more-robust.patch
and users are supposed to use page_is_pfmemalloc(page) instead.

Neither the size nor the layout of the structure has changed as a result but we
have to fake the removed field for the kabi checker. We want to keep pfmemalloc
invisible outside of the kabi checker to force out-of-tree users to recompile
and do the check properly.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

---
 include/linux/mm_types.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -85,6 +85,12 @@ struct page {
 	union {
 		pgoff_t index;		/* Our offset within mapping. */
 		void *freelist;		/* SLUB: freelist req. slab lock */
+#ifdef __GENKSYMS__
+		bool pfmemalloc;	/* pfmemalloc has been removed
+					 * and users are supposed to use
+					 * page_is_pfmemalloc(page) instead
+					 */
+#endif
 	};
 	struct list_head lru;		/* Pageout list, eg. active_list
 					 * protected by zone->lru_lock !
